### PR TITLE
Fixnamespace

### PR DESCRIPTION
**Pull request by evolvedmicrobe**
R packages usually like to avoid any library() calls in the R code,
preferring to use package prefixes (e.g. jsonlite::) along with
IMPORTs, rather than using DEPENDS and a library() call at the start of
each file (see: http://r-pkgs.had.co.nz/namespace.html#namespace), so I
changed this.

Additionally, I added test files directly, and changed some of the code
so the tests will pass.  I am not sure if I am following the schema
though, I will highlight the places I was uncertain about below.

Apologies for the white-space edits, I normally try to leave those out.

### CODE REVIEW

File: R/io.R [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 
 #' General func to load JSON from a file
 #' @export
 loadJsonFromFile <- function(path) {
-  loginfo(paste("Loading tool contract from", path))
+  logging::loginfo(paste("Loading tool contract from", path))
   if (file.exists(path)) {
     s <- readChar(path, file.info(path)$size)
-    d <- fromJSON(s)
+    d <- jsonlite::fromJSON(s, simplifyDataFrame = FALSE)
```
Comment (evolvedmicrobe) [inline R/io.R]: I think jsonlite made an update that changed this option, not having this load as a simplified data frame seemed to help.

File: R/io.R [hunk 2]
```diff
@@ -44,41 +43,44 @@ loadToolContractFromPath <- function(path) {
...
+                                                                                     file_type_id)))
   }
   toOutputType <- function(ft) {
-    return(list(file_type_id = ft@fileTypeId, id = paste("id", ft@fileTypeId, 
-      sep = "_"), title = paste("Display name ", ft@fileTypeId), default_name = paste(ft@baseName, 
-      ft@fileExt, sep = "."), description = paste("File type", ft@fileTypeId)))
+    tmp = toInputType(ft)
+    splitID = strsplit(tmp$file_type_id, "\\.")[[1]]
+    ext = splitID[length(splitID)]
+    tmp$default_name = paste(ft$default_name, ext, sep = ".")
```
Comment (evolvedmicrobe) [inline R/io.R]: This looked a bit strange to me, I gather the ext thing might not exists anymore? The output type in json already had a default name, so I thought I should just grab that but wanted to check.

File: R/io.R [hunk 3]
```diff
@@ -44,41 +43,44 @@ loadToolContractFromPath <- function(path) {
...
   authorComment <- paste("Created from pbcommandR version ", PB_COMMANDR_VERSION)
   isDistributed <- FALSE
   schemaOptions <- list()
   # The 'id' needs to be fixed. ft@fileTypeId_{index} this isn't really used in the
   # R 'quick' model, so it's fine.
   toInputType <- function(ft) {
-    return(list(file_type_id = ft@fileTypeId, id = paste("id", ft@fileTypeId, 
-      sep = "_"), title = paste("Display name ", ft@fileTypeId), description = paste("File type ", 
-      ft@fileTypeId)))
+    file_type_id = ft$file_type_id
```
Comment (evolvedmicrobe) [inline R/io.R]: I assume fileTypeId has been replaced by file_type_id?

Comment (mpkocher) [inline R/io.R]: This looks correct, but I'll take a closer look.

In principle, these schemas should officially define the interface.

https://github.com/PacificBiosciences/pbcommand/tree/master/pbcommand/schemas

**Comment by mpkocher** [comment]
Awesome and thanks for addressing these issues.

Sorry for any churn, there's been a bit of evolution of the TC/RTC model since this code was written.

**Pull request merged**